assembly/fabric8-ide-dashboard-war/Dockerfile

```diff
@@ -1,25 +1,35 @@
-# Copyright (c) 2012-2018 Red Hat, Inc
-# All rights reserved. This program and the accompanying materials
-# are made available under the terms of the Eclipse Public License v1.0
-# which accompanies this distribution, and is available at
-# http://www.eclipse.org/legal/epl-v10.html
+# Copyright (c) 2015-2019 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
 
 # This is a Dockerfile allowing to build dashboard by using a docker container.
-# Build step: $ docker build -t eclipse-che-dashboard
+# Build step: $ docker build -t eclipse-che-dashboard .
 # It builds an archive file that can be used by doing later
 #  $ docker run --rm eclipse-che-dashboard | tar -C target/ -zxf -
 FROM node:8.15.1
 
 ARG SOURCES_DIR
+ARG UPSTREAM_VERSION
 
 RUN apt-get update && \
     apt-get install -y git \
     && apt-get -y clean \
     && rm -rf /var/lib/apt/lists/*
-COPY ${SOURCES_DIR}/package.json /dashboard/
-RUN cd /dashboard && npm i yarn && npx yarn install --ignore-scripts
+RUN yarn global upgrade yarn@1.13.0
+COPY  ${SOURCES_DIR}/package.json /dashboard/
+RUN wget https://raw.githubusercontent.com/eclipse/che/${UPSTREAM_VERSION}/dashboard/yarn.lock -P /dashboard
+COPY  ${SOURCES_DIR}/typings.json /dashboard/
+WORKDIR /dashboard
 COPY ${SOURCES_DIR} /dashboard/
-RUN cd /dashboard && npx yarn
-RUN cd /dashboard && cd target/ && tar zcf /tmp/dashboard.tar.gz dist/
+RUN yarn install --ignore-optional
```

Comment:
I believe that this command should be one line above. It was supposed that yarm install can be cached when deps are not changed, so it is needed to be called before injecting all the sources

Reply:
Can you also copy comments from here https://github.com/redhat-developer/rh-che/pull/1333/commits/b4e7a4860762427abe8e01abd874b38dc128bea8#diff-ff7c231d25ba3487c1cacd45910ae494R21 and here https://github.com/redhat-developer/rh-che/pull/1333/commits/b4e7a4860762427abe8e01abd874b38dc128bea8#diff-ff7c231d25ba3487c1cacd45910ae494R27

Reply:
Done -- good catch.

I had to add `--ignore-scripts` to the yarn command since it was executing `gulp build`.